src/main/kotlin/io/snyk/plugin/ui/jcef/ApplyFixHandler.kt

```diff
@@ -0,0 +1,215 @@
+                            lines.removeAt(originalLineIndex)
+                        }
+                    }
+                    is Change.Context -> {
+                        originalLineIndex++  // Move past unchanged context lines
+                    }
+                }
+            }
+        }
+        return lines.joinToString("\n")
+    }
+
+    fun parseDiff(diff: String): DiffPatch {
+        val lines = diff.lines()
+        val originalFile = lines.first { it.startsWith("---") }.substringAfter("--- ")
+        val fixedFile = lines.first { it.startsWith("+++") }.substringAfter("+++ ")
+
+        val hunks = mutableListOf<Hunk>()
+        var currentHunk: Hunk? = null
+        val changes = mutableListOf<Change>()
+
+        for (line in lines) {
+            when {
+                line.startsWith("@@") -> {
+                    // Parse hunk header (e.g., @@ -4,9 +4,14 @@)
+                    val hunkHeader = line.substringAfter("@@ ").substringBefore(" @@").split(" ")
+                    val original = hunkHeader[0].substring(1).split(",")
+                    val fixed = hunkHeader[1].substring(1).split(",")
+
+                    val startLineOriginal = original[0].toInt()
+                    val numLinesOriginal = original.getOrNull(1)?.toInt() ?: 1
+                    val startLineFixed = fixed[0].toInt()
+                    val numLinesFixed = fixed.getOrNull(1)?.toInt() ?: 1
+
+                    if (currentHunk != null) {
+                        hunks.add(currentHunk.copy(changes = changes.toList()))
+                        changes.clear()
+                    }
+                    currentHunk = Hunk(
+                        startLineOriginal = startLineOriginal,
+                        numLinesOriginal = numLinesOriginal,
+                        startLineFixed = startLineFixed,
+                        numLinesFixed = numLinesFixed,
+                        changes = emptyList()
+                    )
+                }
+                line.startsWith("---") || line.startsWith("+++") -> {
+                    // Skip file metadata lines (--- and +++)
+                    continue
+                }
+                line.startsWith("-") -> changes.add(Change.Deletion(line.substring(1).trim()))
+                line.startsWith("+") -> changes.add(Change.Addition(line.substring(1).trim()))
+                else -> changes.add(Change.Context(line.trim()))
+            }
+        }
+
+        // Add the last hunk
+        if (currentHunk != null) {
+            hunks.add(currentHunk.copy(changes = changes.toList()))
+        }
+
+        return DiffPatch(
+            originalFile = originalFile,
+            fixedFile = fixedFile,
+            hunks = hunks
+        )
+    }
+
+    private fun findVirtualFile(filePath: String): VirtualFile? {
+        return LocalFileSystem.getInstance().findFileByPath(filePath)
```

Comment:
I think you can just use `filePath.toVirtualFile()` here. There should be a helper function in our Utils.kt file.